### DISCUSSION TITLE
How to create a secured sandbox for javascript to run?

### DISCUSSION THREAD

**Discussion by nextfool** [ClearFoundry/ClearScript#303] [Q&A] [answered]
Hi, 
I know that being able to decide which .net object/type been exposed to script is a security feature. I wonder how do I create a even more secure sandbox by:
1. limiting the resources(memory, cpu type, storage, network ... etc)
2. being able to supspend/terminate the script 

I want to create a "guest app runtime", which allows my host app to run any javascript written apps in their separate secured sandbox, guest apps will not aware of each other as well as my host app. My host app will have full control of guest apps.

Can anyone please  give me some basic ideas?  Thank you very much.

**Comment by ClearScriptLib** [comment, marked as answer]
Hi @nextfool,

You can run scripts in separate script engine instances to keep them unaware of each other. However, buggy, malicious, or resource-starved scripts could still crash the process and take down your host.

If you must run untrusted scripts safely, the best way is to run them in separate processes a-la Chromium. That way you could use the capabilities of your operating system to limit resource usage and recover from process crashes.

Good luck!
